### PR TITLE
User config

### PR DESCRIPTION
**Pull request by CJ-Wright**
Support user based configurations. Support alternate branches and remotes.

@scopatz I might need some help with the design for how handle alternate remotes and branches.

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        json.dump({'hello': 'world'}, f)
+
+    backup = sys.stdout
+    sys.stdout = StringIO()
+    main(['rc'])
+    out2 = sys.stdout.getvalue()
+    sys.stdout.close()
+    sys.stdout = backup
+    assert out1 != out2
+    assert 'hello' in out2
```
Comment (scopatz) [inline tests/test_main.py]: Need newline at end of file.

**Comment by scopatz** [comment]
Looks like the tests are failing for some reason

**Comment by CJ-Wright** [comment]
Yea there is something wrong in the set up with xonsh on this side.

File: requirements/run.txt [hunk 2]
```diff
@@ -3,3 +3,4 @@ ruamel.yaml
 docutils
 jinja2
 cerberus
+xonsh
```
Comment (scopatz) [inline requirements/run.txt]: Looks like you are importing from rever, but don't have it as a dependency

Comment (CJ-Wright) [reply requirements/run.txt]: Ah yes that could be a problem!

**Comment by CJ-Wright** [comment]
@scopatz even fixing that there seems to be an issue with the finding of the database module. Am I messing up the setup.py?

**Comment by scopatz** [comment]
database.xsh is getting installed correctly: https://travis-ci.org/regro/regolith/builds/390807757#L781

You need to add the import hooks before you can import xonsh modules, though: http://xon.sh/api/imphooks.html#xonsh.imphooks.install_import_hooks

**Comment by CJ-Wright** [comment]
Ah ok thank you very much!

File: regolith/database.xsh [hunk 3]
```diff
@@ -28,12 +30,15 @@ def load_git_database(db, client, rc):
     dbdir = dbdirname(db, rc)
     # get or update the database
     if os.path.isdir(dbdir):
-        cmd = ['git', 'pull']
-        cwd = dbdir
+        with indir(dbdir):
+            git pull upstream master or git pull origin master or git pull @(rc.remote) master or git pull @(rc.remote) @(rc.branch) or git pull
```
Comment (CJ-Wright) [inline regolith/database.xsh]: @scopatz what is the best way to get this over multiple lines?

Comment (scopatz) [reply regolith/database.xsh]: backslash at the end of the line (line continuation)

Comment (CJ-Wright) [reply regolith/database.xsh]: Ah ok I was trying to do 
```
(git pull upstream master
 or git pull origin master ...)
```

Comment (scopatz) [reply regolith/database.xsh]: Or you could use `()` and explicit subprocesses, eg. `(![git pull upstream master] or\n ![git pull origin master])`

Comment (scopatz) [reply regolith/database.xsh]: Yeah, you can, you just need to be explicit about it, I think

Comment (CJ-Wright) [reply regolith/database.xsh]: Xonsh seems not happy about
```
git pull upstream master \
            or git pull origin master \
            or git pull @(rc.remote) master \
            or git pull @(rc.remote) @(rc.branch) \
            or git pull
```

Comment (scopatz) [reply regolith/database.xsh]: Is the backslash the last char  on the line (no whitespace)?

Comment (CJ-Wright) [reply regolith/database.xsh]: Yes

Comment (scopatz) [reply regolith/database.xsh]: hmmm please report on the xonsh issue tracker and use explicit subprocs instead. Thanks!

**Comment by CJ-Wright** [comment]
This needs docs somewhere. And I want to make certain that this is the behavior we want.

**Comment by scopatz** [comment]
Code looks good to me!

**Comment by CJ-Wright** [comment]
@scopatz do you want to look at the addition to the docs or should I merge this?

**Comment by scopatz** [comment]
LGTM!

**Pull request merged**